spring-kafka/src/main/java/org/springframework/kafka/support/converter/BatchMessageConverter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.support.converter;
+
+import java.lang.reflect.Type;
+import java.util.List;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+import org.springframework.kafka.support.Acknowledgment;
+import org.springframework.messaging.Message;
+
+/**
+ * A Kafka-specific {@link Message} converter strategy.
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Gary Russell
+ */
+public interface BatchMessageConverter<K, V> extends MessageConverter {
```

Comment:
Ugh - I hate `rawTypes` but I guess it's better than adding generics for no reason.